frontend/pages/SoftwarePage/SoftwareTitleDetailsPage/EditSoftwareModal/EditSoftwareModal.tsx

```diff
@@ -161,42 +176,122 @@ const EditSoftwareModal = ({
+
+      renderFlash(
+        "success",
+        <>
+          Successfully edited <b>{formData.selectedApp?.name}</b>.
+          {formData.selfService
+            ? " The end user can install from Fleet Desktop."
+            : ""}
+        </>
+      );
+      onExit();
+      refetchSoftwareTitle();
+    } catch (e) {
+      renderFlash("error", getErrorMessage(e, software as IAppStoreApp));
     }
     setIsUpdatingSoftware(false);
   };
 
-  const onEditSoftware = (formData: IPackageFormData) => {
-    // Check for changes to conditionally confirm save changes modal
-    const updates = deepDifference(formData, {
-      software,
-      installScript: software.install_script || "",
-      preInstallQuery: software.pre_install_query || "",
-      postInstallScript: software.post_install_script || "",
-      uninstallScript: software.uninstall_script || "",
+  const onClickSaveVpp = async (formData: ISoftwareVppFormData) => {
+    const currentData = {
       selfService: software.self_service || false,
-      installType: getInstallType(software),
       targetType: getTargetType(software),
       customTarget: getCustomTarget(software),
       labelTargets: generateSelectedLabels(software),
-    });
+    };
 
-    setPendingUpdates(formData);
+    setPendingVppUpdates(formData);
 
-    const onlySelfServiceUpdated =
-      Object.keys(updates).length === 1 && "selfService" in updates;
-    if (onlySelfServiceUpdated) {
-      // Proceed with saving changes (API expects only changes)
-      onSaveSoftwareChanges(formData);
+    const updates = deepDifference(formData, currentData);
+
+    if (isOnlySelfServiceUpdated(updates)) {
+      onEditVpp(formData);
     } else {
-      // Open the confirm save changes modal
       setShowConfirmSaveChangesModal(true);
     }
   };
 
-  const onConfirmSoftwareChanges = () => {
+  const onClickConfirmChanges = () => {
     setShowConfirmSaveChangesModal(false);
-    onSaveSoftwareChanges(pendingUpdates);
+    if (installerType === "package") {
+      onEditPackage(pendingPackageUpdates);
+    } else {
+      onEditVpp(pendingVppUpdates);
+    }
+  };
+
+  const renderForm = () => {
+    if (installerType === "package") {
+      const softwarePackage = software as ISoftwarePackage;
+      return (
+        <PackageForm
+          labels={labels ?? []}
+          className={`${baseClass}__package-form`}
+          isEditingSoftware
+          onCancel={onExit}
+          onSubmit={onClickSavePackage}
+          defaultSoftware={software}
+          defaultInstallScript={softwarePackage.install_script}
+          defaultPreInstallQuery={softwarePackage.pre_install_query}
+          defaultPostInstallScript={softwarePackage.post_install_script}
+          defaultUninstallScript={softwarePackage.uninstall_script}
+          defaultSelfService={softwarePackage.self_service}
+        />
+      );
+    }
+
+    return (
+      <SoftwareVppForm
+        labels={labels || []}
```

Comment:
Any reason for using `||` here rather than `??` as on L272 (in `<PackageForm>`)?

Reply:
I only see `labels={labels || []}` in `TargetLabelSelector` of `<PackageForm>`, can you double check, or clarify what you're referring to with the `??`

Reply:
This one: https://github.com/fleetdm/fleet/pull/25815/files#diff-8a0222bfd0a40e01b432d64ea671889b02048209722220b546bc7ae7e1022415R272

Reply:
Ahh, 272 on this file, I was looking at PackageForm file, will update, ty!